// eslint-plugin-office-addins/lib/main.js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const rules_1 = __importDefault(require("./rules"));
module.exports = {
    rules: rules_1.default,
    configs: {
        recommended: {
            parser: "@typescript-eslint/parser",
            plugins: ["@typescript-eslint", "office-addins", "prettier"],
            parserOptions: {
                ecmaVersion: 6,
                sourceType: "module",
                ecmaFeatures: {
                    jsx: true,
                },
                project: "./tsconfig.json",
            },
            extends: ["eslint:recommended"],
            rules: {
                "prettier/prettier": ["error", { endOfLine: "auto" }],
                "no-eval": "error",
                "no-delete-var": "warn",
                "no-octal": "warn",
                "no-inner-declarations": "warn",
            },
        },
        react: {
            extends: ["plugin:office-addins/recommended", "plugin:react/recommended"],
        },
        reactnative: {
            extends: ["plugin:office-addins/recommended", "plugin:react-native/all"],
        },
    },
};
